src/analyses/mallocWrapperAnalysis.ml

```diff
@@ -55,34 +55,39 @@ struct
   let threadspawn ctx lval f args fctx = ctx.local
   let exitstate  v = D.top ()
 
+
   (* refs to reassign unmarshaled in init *)
   let heap_hash = ref (Hashtbl.create 113)
   let heap_vars = ref (Hashtbl.create 113)
 
   type marshal = {
-    heap_hash: (location, varinfo) Hashtbl.t;
+    heap_hash: (string, varinfo) Hashtbl.t;
```

Comment:
A string-based hashtable is not the best. It'd be better if the keys were `Node.t` directly and no `nodeId` string conversion would be necessary.

To make that work properly, we shouldn't use the polymorphic `Hashtbl` (which would use polymorphic comparisons on nodes), but rather something like
```ocaml
module NH = Hashtbl.Make (Node)
(* ... *)
heap_hash: varinfo NH.t
```

I've thought about refactoring it here before, but there wasn't a real need before, although now would be a good time.
Then the other hashtable could also be properly made to use `CilType.Varinfo` keys directly instead of `int`.